stdlib/src/memory/memory.mojo

```diff
@@ -50,9 +50,10 @@ fn _align_down(value: Int, alignment: Int) -> Int:
 
 
 @always_inline
-fn _memcmp_impl_unconstrained[
+fn _memcmp_impl[
     type: DType
 ](s1: UnsafePointer[Scalar[type], _], s2: __type_of(s1), count: Int) -> Int:
+    constrained[type.is_unsigned(), "the input dtype must be unsigned"]()
```

Comment:
Since we use `SIMD.__gt__` etc. in this function, if we don't restrict to `unsigned`, the result will be wrong.